### PR TITLE
Add resource limits to telemeter server and prometheus

### PR DESCRIPTION
**Pull request by jfchevrette**
- Add resource requests and limits variables to telemeter server
- Add resource requests and limits variables to prometheus
- Allow setting the resource requests and limits via the templates

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
It’s unclear if this is happening from my review, but please do not set a cpu limit.  Cpu requests should be sufficient.

**Comment by s-urbaniak** [comment]
@derekwaynecarr the limits configured here do not refer to components running inside cluster installations. This configuration rather refers to components running on our premises (telemeter backend), supervisioned by service delivery. According to @jfchevrette limits are needed here for capacity management.

**Comment by jfchevrette** [comment]
@derekwaynecarr can you elaborate on why you recommend not setting a cpu limit? 

From an SRE point of view, we find limits valuable as this allows us to control how much services can consume and prevent cluster saturation and alleviate the 'bad neighbor' effect. There is much less guesswork involved in capacity planning when we have cpu and memory limits on all services we run.

File: jsonnet/telemeter/server.libsonnet [hunk 1]
```diff
@@ -18,6 +19,14 @@ local list = import 'lib/list.libsonnet';
       elideLabels+: [
         'prometheus_replica',
       ],
+      // resourceLimits: {
```
Comment (s-urbaniak) [inline jsonnet/telemeter/server.libsonnet]: do you mind to remove those comments?

Comment (jfchevrette) [reply jsonnet/telemeter/server.libsonnet]: removed in 2032d64 :+1:

File: manifests/prometheus/prometheus.yaml [hunk 2]
```diff
@@ -43,6 +43,7 @@ spec:
     beta.kubernetes.io/os: linux
   replicas: 2
   resources:
+    limits: {}
```
Comment (s-urbaniak) [inline manifests/prometheus/prometheus.yaml]: @jfchevrette i didn't verify, but is this fine?

File: manifests/server/statefulSet.yaml [hunk 3]
```diff
@@ -99,6 +99,9 @@ spec:
             path: /healthz/ready
             port: 8443
             scheme: HTTPS
+        resources:
+          limits: {}
+          requests: {}
```
Comment (s-urbaniak) [inline manifests/server/statefulSet.yaml]: see above

File: manifests/prometheus/prometheus.yaml (hunk 2 above)
Comment (jfchevrette) [reply manifests/prometheus/prometheus.yaml]: I've tested that empty map is valid and the end result is that no limits are set. This is the same as before.

File: manifests/server/statefulSet.yaml (hunk 3 above)
Comment (jfchevrette) [reply manifests/server/statefulSet.yaml]: I've tested that empty map is valid and the end result is that no limits are set. This is the same as before.

**Pull request merged**